summarize.py

```diff
@@ -1,37 +1,57 @@
 from langchain.document_loaders import PyPDFLoader
 from langchain.text_splitter import CharacterTextSplitter
-from langchain.embeddings.openai import OpenAIEmbeddings
-from langchain.vectorstores import FAISS
-from langchain.chains import RetrievalQA
 from langchain import OpenAI
-
-
-def summarize(pdf_path):
+from langchain.chains.summarize import load_summarize_chain
+from time import time
+import argparse
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    "-f",
+    "--file_path",
+    type=str,
+    help="Path to the PDF file to summarize",
+    required=True,
+)
+parser.add_argument(
+    "-c",
+    "--chain_type",
+    type=str,
+    help='Open ai summary chain_type: "stuff", "map_reduce" or "refine"',
+    default="map_reduce",
+)
+
+args = parser.parse_args()
+
+
+def summarize(pdf_path, chain_type):
     # Load the PDF
     print(f"Loading PDF from path: {pdf_path}")
-    loader = PyPDFLoader(file_path = pdf_path)
-
-    prompt = """Give me a 2 sentence summary of the following text: \n"""
+    loader = PyPDFLoader(file_path=pdf_path)
 
     # Split the PDF into chunks of text
-    text_splitter = CharacterTextSplitter(chunk_size=500, chunk_overlap=100, separator="/n")
+    text_splitter = CharacterTextSplitter(
+        chunk_size=750, chunk_overlap=100, separator="/n"
+    )
     document = loader.load()
-    docs = text_splitter.split_documents(documents = document)
-
-    # Create a vectorstore from the documents to use for retrieval
-    embeddings = OpenAIEmbeddings()
-    vectorstore =  FAISS.from_documents(docs, embeddings)
```

Comment:
Have you noticed FAISS being better in any way over other vector databases you've used? e.g. does it return better results? is it faster? etc.

Reply:
Short answer, I don't know. 
I haven't tried chroma at this point, I found the FAISS stuff easy to use (and included in the langchain library version I had originally. I re-watched the video from the course I did to try to figure out a comparison. No luck. Duck duck go (...) returned me this result: 
https://harishgarg.com/writing/best-vector-databases-for-ai-apps/ 
with this tidbit, which was interesting (Chroma was #1) so they might be different strategies entirely. 
Faiss by Facebook
Not a vector database but a library for efficient similarity search and clustering of dense vectors. It’s [open source](https://github.com/facebookresearch/faiss).
Did not answer your question. Good luck! this is on my back burner of things to think about.